backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/MoveOrCopyDiskCommand.java

```diff
@@ -173,7 +173,17 @@ && checkCanBeMoveInVm()
                 && checkIfNeedToBeOverride()
                 && setAndValidateDiskProfiles()
                 && setAndValidateQuota()
-                && validatePassDiscardSupportedForDestinationStorageDomain();
+                && validatePassDiscardSupportedForDestinationStorageDomain()
+                && isCopyWithLocalDc();
+    }
+
+    protected boolean isCopyWithLocalDc() {
+        if(getStoragePool().isLocal() && getStorageDomain().getStorageType() == StorageType.MANAGED_BLOCK_STORAGE
+                || getStorageDomain().getStorageType() == StorageType.MANAGED_BLOCK_STORAGE && getStoragePool().isLocal())  {
```

Comment:
you're checking the same thing here twice, no? you probably meant checking the source and destination domains?